### PR TITLE
Add support for tensor product elements (quads and hexes)

### PR DESCRIPTION
**Pull request by MTCam**

### CODE REVIEW

File: y3prediction/prediction.py [hunk 1]
```diff
@@ -1982,6 +1992,12 @@ def compute_smoothed_char_length(href_fluid, comm_ind):
                  for dd_bdry in filter_part_boundaries(
                      dcoll, volume_dd=dd_vol_fluid, neighbor_volume_dd=dd_vol_wall)})
 
+        # smooth_href_fluid_rhs = diffusion_operator(
+        #    dcoll, smoothness_diffusivity, fluid_smoothness_boundaries,
+        #    href_fluid, char_length_fluid,
+        #    quadrature_tag=quadrature_tag, dd=dd_vol_fluid,
+        #    comm_tag=(_SmoothCharDiffFluidCommTag, comm_ind))*current_dt
+
```
Comment (MTCam) [inline y3prediction/prediction.py:2000]:
```suggestion
```

File: buildMirge.sh [hunk 2]
```diff
@@ -1,9 +1,11 @@
 #!/bin/bash
 
 # default branch for building mirgecom for this driver
-mirge_branch="production"
+mirge_branch="tensor-product-elements"
```
Comment (MTCam) [inline buildMirge.sh:4]: Not this one.

File: y3prediction/prediction.py [hunk 3]
```diff
@@ -449,17 +449,24 @@ def main(actx_class,
...
+    use_tensor_product_elements = configurate("use_tensor_product_elements",
+                                              input_data, False)
+    if use_tensor_product_elements:
+        from grudge.array_context import TensorProductMPIFusionContractorArrayContext
+        actx = initialize_actx(TensorProductMPIFusionContractorArrayContext, comm)
+    else:
+        actx = initialize_actx(actx_class, comm)
+    queue = getattr(actx, "queue", None)
+    use_profiling = actx_class_is_profiling(actx_class)
+    alloc = getattr(actx, "allocator", None)
```
Comment (MTCam) [inline y3prediction/prediction.py:468]:
```suggestion
    from mirgecom.array_context import initialize_actx, actx_class_is_profiling
    actx = initialize_actx(actx_class, comm)
    queue = getattr(actx, "queue", None)
    use_profiling = actx_class_is_profiling(actx_class)
    alloc = getattr(actx, "allocator", None)
```

**Comment by anderson2981** [comment]
merged with previous changes.

**Pull request closed without merging**